phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4006,6 +4037,33 @@ private PhoenixConnection upgradeSystemTask(PhoenixConnection metaConnection)
                         "ALTER TABLE " + taskTableFullName + " SET " + TTL + "=" + TASK_TABLE_TTL);
                 clearCache();
             }
+            // If SYSTEM.TASK does not have disabled regions split policy,
+            // set it up here while upgrading it
+            try (Admin admin = metaConnection.getQueryServices().getAdmin()) {
```

Comment:
Is it possible to add a test case inside BackwardsCompatibilityIT to make sure that the upgrade path changes work and SYSTEM.TASK comes up with the expected split policy?

Reply:
Added a test on `BackwardsCompatibilityIT`. Tested on 4.x with compatible client versions: 4.14.3 and 4.15.0.
For one, split policy was set by DDL statement at server and for another, upgrade path was followed to ensure split policy is updated.